src/org/thoughtcrime/securesms/service/GenericForegroundService.java

```diff
@@ -47,6 +49,7 @@
   private static final AtomicInteger NEXT_ID = new AtomicInteger();
   private static final AtomicBoolean CHANNEL_CREATED = new AtomicBoolean(false);
 
+  private static int startedCounter = 0;
```

Comment:
why don't you use a bool here? It's never reset to 0 as far as I can see.

Reply:
it is increased on adding a foreground-service and decreased on removing one (in theory, there could be several foreground-services) if it is 0 again, we know, there is nothing ongoing. a simple boolean would not allow using several foreground-services.